extensions/ql-vscode/src/databases-ui.ts

```diff
@@ -378,21 +378,28 @@ export class DatabaseUI extends DisposableObject {
 
   handleRemoveOrphanedDatabases = async (): Promise<void> => {
     logger.log('Removing orphaned databases from workspace storage.');
-    let dbDirs =
-      // read directory
-      (await fs.readdir(this.storagePath, { withFileTypes: true }))
-        // remove non-directories
-        .filter(dirent => dirent.isDirectory())
-        // get the full path
-        .map(dirent => path.join(this.storagePath, dirent.name))
-        // remove databases still in workspace
-        .filter(dbDir => {
-          const dbUri = Uri.file(dbDir);
-          return this.databaseManager.databaseItems.every(item => item.databaseUri.fsPath !== dbUri.fsPath);
-        });
-
-    // remove non-databases
-    dbDirs = await asyncFilter(dbDirs, isLikelyDatabaseRoot);
+    let dbDirs = undefined;
+
+    try {
+      dbDirs =
+        // read directory
+        (await fs.readdir(this.storagePath, { withFileTypes: true }))
+          // remove non-directories
+          .filter(dirent => dirent.isDirectory())
+          // get the full path
+          .map(dirent => path.join(this.storagePath, dirent.name))
+          // remove databases still in workspace
+          .filter(dbDir => {
+            const dbUri = Uri.file(dbDir);
+            return this.databaseManager.databaseItems.every(item => item.databaseUri.fsPath !== dbUri.fsPath);
+          });
+
+      // remove non-databases
+      dbDirs = await asyncFilter(dbDirs, isLikelyDatabaseRoot);
+    } catch (e) {
+      logger.log(`Not able to find the storage path '${e.message}'. Ignoring request to remove orphaned database directories.`);
```

Comment:
Minor: You could do a check up front with `await fs.exists(this.storagePath)` so that we can distinguish between the storage path not existing, and some other failure to read its contents.
```suggestion
      logger.log(`Not able to read from the storage path '${e.message}'. Ignoring request to remove orphaned database directories.`);
```

Reply:
Let me do that and fix it up a bit more.

Reply:
OK. I removed the try/catch around collecting the `dbDirs` since if there is an error at this point, that indicates something much stranger going on that might actually be important for the user.